src/components/task-plan/plan-mixin.coffee

```diff
@@ -53,14 +53,15 @@ module.exports =
     # The logic here is this way because we need to be able to add an invalid
     # state to the form.  Blame @fredasaurus
     if (saveable)
-      TaskPlanStore.addChangeListener(@saved)
+      TaskPlanActions.saved.addListener(@saved)
       TaskPlanActions.save(id)
     else
       @setState({invalid: true})
 
   saved: ->
     courseId = @props.courseId
-    TaskPlanStore.removeChangeListener(@saved)
+    TaskPlanStore.saved.removeListener(@saved)
```

Comment:
I think that if `save` ends up erroring it will leave a dangling listener (unless you listen to `FAILED` as well)